enterprise/server/cmd/ci_runner/main.go

```diff
@@ -2272,3 +2285,50 @@ func runCredentialHelper() error {
 		return nil
 	}
 }
+
+// Attempts to free up disk space if necessary.
+func reclaimDiskSpace(ctx context.Context) error {
+	if runtime.GOOS != "linux" {
+		// Only do this on Linux VMs for now.
+		return nil
+	}
+	usage, err := diskUsageFraction()
+	if err != nil {
+		return fmt.Errorf("get disk usage: %w", err)
+	}
+	if usage < highDiskUsageThreshold {
+		return nil
+	}
+
+	// We should be in the git repo root at this point - run git gc.
```

Comment:
Previously we discussed running git gc at the end of every run. That should amortize the costs and make the behavior more predictable. WHat do you think of that approach instead?

Reply:
I'm not sure if the cost is actually amortized - if I run `git gc` in my local repo, it always seems to do a lot of work.